All CI jobs failing due to some change affecting JS compilation

@mattpap all CI jobs including previously passing ones, are now failing with this:
```
=================================== FAILURES ===================================
_______________ test_functickformatter_from_coffeescript_no_arg ________________
    def test_functickformatter_from_coffeescript_no_arg():
        coffee_code = dedent("""
            square = (x) -> x * x
            return square(tick)
            """)
    
        formatter = FuncTickFormatter.from_coffeescript(code=coffee_code)
    
>       assert formatter.code == dedent("""\
            "use strict";
            var square;
            square = function (x) {
                return x * x;
            };
            return square(tick);
            """)
E       assert '"use strict"...quare(tick);\n' == '"use strict";...uare(tick);\n'
E           "use strict";
E         - Object.defineProperty(exports, "__esModule", { value: true });
E           var square;
E           square = function (x) {
E               return x * x;
E           };
E           return square(tick);
bokeh/models/tests/test_formatters.py:60: AssertionError
______________ test_functickformatter_from_coffeescript_with_args ______________
    def test_functickformatter_from_coffeescript_with_args():
        coffee_code = dedent("""
             return slider.get("value") // 2 + tick
             """)
    
        slider = Slider()
        formatter = FuncTickFormatter.from_coffeescript(code=coffee_code, args={"slider": slider})
    
>       assert formatter.code == dedent("""\
            "use strict";
            return Math.floor(slider.get("value") / 2) + tick;
            """)
E       assert '"use strict".../ 2) + tick;\n' == '"use strict";... 2) + tick;\n'
E           "use strict";
E         - Object.defineProperty(exports, "__esModule", { value: true });
E           return Math.floor(slider.get("value") / 2) + tick;
bokeh/models/tests/test_formatters.py:78: AssertionError
_______________________ test_nodejs_compile_coffeescript _______________________
    def test_nodejs_compile_coffeescript():
>       assert nodejs_compile("""(a, b) -> a + b""", "coffeescript", "some.coffee") == \
            dict(code=""""use strict";\n(function (a, b) {\n    return a + b;\n});\n""", deps=[])
E       assert {u'code': u'"..., u'deps': []} == {'code': '"use...', 'deps': []}
E         Omitting 1 identical items, use -v to show
E         Differing items:
E         {'code': '"use strict";\nObject.defineProperty(exports, "__esModule", { value: true });\n(function (a, b) {\n    return a + b;\n});\n'} != {'code': '"use strict";\n(function (a, b) {\n    return a + b;\n});\n'}
E         Use -v to get the full diff
bokeh/util/tests/test_compiler.py:6: AssertionError
________________________ test_nodejs_compile_javascript ________________________
    def test_nodejs_compile_javascript():
>       assert nodejs_compile("""function f(a, b) { return a + b; };""", "javascript", "some.js") == \
            dict(code=""""use strict";\nfunction f(a, b) { return a + b; }\n;\n""", deps=[])
E       assert {u'code': u'"..., u'deps': []} == {'code': '"use...', 'deps': []}
E         Omitting 1 identical items, use -v to show
E         Differing items:
E         {'code': '"use strict";\nObject.defineProperty(exports, "__esModule", { value: true });\nfunction f(a, b) { return a + b; }\n;\n'} != {'code': '"use strict";\nfunction f(a, b) { return a + b; }\n;\n'}
E         Use -v to get the full diff
bokeh/util/tests/test_compiler.py:41: AssertionError
============================= 207 tests deselected =============================
======= 4 failed, 927 passed, 207 deselected, 20 rerun in 60.78 seconds ========

```
The version of nodejs (`6.6.0`) is installed from the `bokeh` conda channel so I don't see how that could have changed. Not sure what else could have updated to start causing these failures.

--- a/bokeh/models/tests/test_formatters.py
+++ b/bokeh/models/tests/test_formatters.py
@@ -59,6 +59,7 @@ def test_functickformatter_from_coffeescript_no_arg():
 
     assert formatter.code == dedent("""\
         "use strict";
+        Object.defineProperty(exports, "__esModule", { value: true });
         var square;
         square = function (x) {
             return x * x;
@@ -77,6 +78,7 @@ def test_functickformatter_from_coffeescript_with_args():
 
     assert formatter.code == dedent("""\
         "use strict";
+        Object.defineProperty(exports, "__esModule", { value: true });
         return Math.floor(slider.get("value") / 2) + tick;
         """)
     assert formatter.args == {"slider": slider}

--- a/bokeh/util/tests/test_compiler.py
+++ b/bokeh/util/tests/test_compiler.py
@@ -4,10 +4,21 @@ from bokeh.util.compiler import nodejs_compile
 
 def test_nodejs_compile_coffeescript():
     assert nodejs_compile("""(a, b) -> a + b""", "coffeescript", "some.coffee") == \
-        dict(code=""""use strict";\n(function (a, b) {\n    return a + b;\n});\n""", deps=[])
+        dict(code="""\
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+(function (a, b) {
+    return a + b;
+});
+""", deps=[])
 
     assert nodejs_compile("""some = require 'some/module'""", "coffeescript", "some.coffee") == \
-        dict(code=""""use strict";\nvar some;\nsome = require('some/module');\n""", deps=["some/module"])
+        dict(code="""\
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+var some;
+some = require('some/module');
+""", deps=["some/module"])
 
     assert nodejs_compile("""(a, b) -> a + b +""", "coffeescript", "some.coffee") == \
         dict(error=dict(
@@ -39,10 +50,19 @@ def test_nodejs_compile_coffeescript():
 
 def test_nodejs_compile_javascript():
     assert nodejs_compile("""function f(a, b) { return a + b; };""", "javascript", "some.js") == \
-        dict(code=""""use strict";\nfunction f(a, b) { return a + b; }\n;\n""", deps=[])
+        dict(code="""\
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+function f(a, b) { return a + b; }
+;
+""", deps=[])
 
     assert nodejs_compile("""var some = require('some/module');""", "javascript", "some.js") == \
-        dict(code=""""use strict";\nvar some = require('some/module');\n""", deps=["some/module"])
+        dict(code="""\
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+var some = require('some/module');
+""", deps=["some/module"])
 
     assert nodejs_compile("""function f(a, b) { eturn a + b; };""", "javascript", "some.js") == \
         dict(error=dict(

--- a/bokehjs/package.json
+++ b/bokehjs/package.json
@@ -59,7 +59,7 @@
     "event-stream": "^3.3.1",
     "jsdom": "^9.8.3",
     "websocket":"^1.0.22",
-    "typescript": "^2.0.3",
+    "typescript": "^2.2.1",
     "coffee-script": "^1.10.0",
     "detective": "^4.3.1",
     "jslint": "^0.10.3",

